assembl/static2/js/app/components/home/video.jsx

```diff
@@ -2,6 +2,60 @@ import React from 'react';
 import { connect } from 'react-redux';
 import { Translate } from 'react-redux-i18n';
 import { Grid, Row, Col } from 'react-bootstrap';
+import get from 'lodash/get';
+
+// https://en.wikipedia.org/wiki/Video_file_format
+const videoExtensions = [
+  'webm',
+  'mkv',
+  'flv',
+  'vob',
+  'ogv',
+  'ogg',
+  'drc',
+  'gif',
+  'gifv',
+  'mng',
+  'avi',
+  'mov',
+  'qt',
+  'wmv',
+  'yuv',
+  'rm',
+  'rmvb',
+  'asf',
+  'amv',
+  'mp4',
+  'm4p',
+  'm4v',
+  'mpg',
+  'mp2',
+  'mpeg',
+  'mpe',
+  'mpv',
+  'mpg',
+  'mpeg',
+  'm2v',
+  'm4v',
+  'svi',
+  '3gp',
+  '3g2',
+  'mxf',
+  'roq',
+  'nsv',
+  'f4v',
+  'f4p',
+  'f4a',
+  'f4b'
+];
+
+const EDFHacks = {
```

Comment:
This part is not EDF-specific, though it was done for them. So EDF should not be part of the name.
We can and should do better to check file vs frame, but this is still a vast improvement from the previous code; so I'm not even sure it should be called a hack as such. (So how about NaiveVideoFileFrameDetector?)
In other words, this should be written in a style that can go to develop.

I had to redo the work for media; ideally there should have been a Media component which can receive the video url and poster as props, and the nature of the url (video/frame/autodetect, with autodetect as default?) and is shared between home and thematic.
But that needs to be scheduled properly.